### ISSUE TITLE
Android / NDK

### ISSUE THREAD

**Issue by clauderobi** [KjellKod/g3log#350]
It is actually more a question at this stage.

Is g3log supported on Android? I see some issues being reported here as well as a PR but it is not clear if it possible to compile under NDK. I was not successful and before going any further I wonder if it hopeless.

**Comment by KjellKod** [comment]
Not sure what issues and PRs you are referring to in terms of Android. This one? https://github.com/KjellKod/g3log/issues/32

The only fork for Android that Im aware of was made in 2015 but it doesn't look like it's actively maintained. 

I have personally not tested g3log on Android, or done any Android development either.  What specifically are the pain points that you are encountering?

**Comment by clauderobi** [comment]
This one https://github.com/KjellKod/g3log/pull/67#issue-52918541

The very first problem was that the definition for string was not found. I found in Build.cmake that `-stdlib=libstdc++`  is set on the compiler CXXFLAGS, which causes NDK to use an incomplete STD lib. (As per https://android.googlesource.com/platform/ndk/+/master/docs/BuildSystemMaintainers.md#STL)

I set CMAKE_SYSTEM_NAME to "Android" so that now the flag is calling libc++ (instead of libstdc++). 

But now, I get execinfo.h not found. To try to resolve this, I tried to set ENABLE_FATAL_SIGNALHANDLING to "OFF" but did not get the result I was hoping for.... (I was hoping that crashhandler_xx would not be included in the build)

This is when I found the PR that I mention above.

**Comment by KjellKod** [comment]
Got it. That PR was opened and closed almost immediately. It seems he just copied the existing code and renamed it. 

I commented here on a way forward for Android: https://github.com/KjellKod/g3log/issues/32#issuecomment-127023721

This was a long time ago, I havent' looked into it since but I'd probably recommend the same thing. 

1. Create a fork of g3log. 
2. Add android specific crash handling since nix-common libraries aren't available. 
3. Use CallStack.h and the simplified approach instead of the more complex approach currently used for Unix.  OR use another approach. The CallStack.h approach was suggested in 2018. There might be much newer approaches available now. I found this thread that might be of interest for implementing the crashhandler_android.cpp. --> https://stackoverflow.com/questions/8115192/android-ndk-getting-the-backtrace#28858941


Then using
